java/core/src/main/java/com/lancedb/lance/schema/LanceField.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.lancedb.lance.schema;
+
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.DictionaryEncoding;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class LanceField {
+
+  private final int id;
+  private final String name;
+  private final boolean nullable;
+  private final ArrowType type;
```

Comment:
similar comment, I think we need a `logicalType` and a class `LogicalType`, and logical type should have a field `name` (corresponding to the unnamed string in `lance_core::datatypes::LogicalType`) to record the string type name.

Reply:
> Yeah, I found map and union types are not supported yet. Are they in the plan?

I think we will add them when someone needs them.  Map will be easy.  Union is a little more tricky but we can support.

Reply:
If we want to extend Arrow we can use Arrow's builtin extension type mechanism.